adls/include/adls_client.h

```diff
@@ -0,0 +1,336 @@
+
+        /// <summary>
+        /// Marks the filesystem for deletion.
+        /// </summary>
+        /// <param name="filesystem">The filesystem name.</param>
+        AZURE_STORAGE_ADLS_API void delete_filesystem(const std::string& filesystem);
+
+        /// <summary>
+        /// Returns whether a filesystem already exists.
+        /// </summary>
+        /// <param name="filesystem">The filesystem name.</param>
+        /// <returns><c>true</c> if the filesystem exists, <c>false</c> otherwise.</returns>
+        AZURE_STORAGE_ADLS_API bool filesystem_exists(const std::string& filesystem);
+
+        /// <summary>
+        /// Sets properties for the filesystem.
+        /// </summary>
+        /// <param name="filesystem">The filesystem name.</param>
+        /// <param name="properties">Key-value pairs of properties</param>
+        AZURE_STORAGE_ADLS_API void set_filesystem_properties(const std::string& filesystem, const std::vector<std::pair<std::string, std::string>>& properties);
+
+        /// <summary>
+        /// Gets properties for the filesystem.
+        /// </summary>
+        /// <param name="filesystem">The filesystem name.</param>
+        /// <returns>Key-value pairs of properties.</returns>
+        AZURE_STORAGE_ADLS_API std::vector<std::pair<std::string, std::string>> get_filesystem_properties(const std::string& filesystem);
+
+        /// <summary>
+        /// Lists filesystems.
+        /// </summary>
+        /// <param name="prefix">Filter results to filesystems within the specified prefix.</param>
+        /// <param name="continuation_token">
+        /// The number of filesystems returned with each invocation is limited. If the number of filesystems to be returned
+        /// exceeds this limit, a continuation token is returned in the response
+        /// When a continuation token is returned in the response, it must be specified in a subsequent invocation of the
+        /// list operation to continue listing the filesystems.
+        /// </param>
+        /// <param name="max_results">An optional value that specifies the maximum number of items to return.</param>
+        /// <returns>A <see cref="azure::storage_adls::list_filesystems_result"> object which contains filesystems list and continuation token.</returns>
+        AZURE_STORAGE_ADLS_API list_filesystems_result list_filesystems_segmented(const std::string& prefix, const std::string& continuation_token = std::string(), const int max_results = 0);
+
+        /// <summary>
+        /// Creates a directory in the filesystem. If the directory already exists, it's overwritten.
+        /// </summary>
+        /// <param name="filesystem">The filesystem name.</param>
+        /// <param name="directory">The directory path.</param>
+        AZURE_STORAGE_ADLS_API void create_directory(const std::string& filesystem, const std::string& directory);
+
+        /// <summary>
+        /// Deletes the directory.
```

Comment:
>Deletes the directory [](start = 12, length = 21)

Is the behavior different with `delete_filesystem`? If not, why is the wording different?

Reply:
They're different in REST API documentation.